File: backend/jeerai-backend/src/main/java/com/jeerai/backend/model/Issue.java
package com.jeerai.backend.model;
import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Issue {
    private String id;
    private String key;
    private String title;
    private String status;
    private String priority;
    private User assignee;
    private User reporter;
    private Instant createdAt;
    private Instant updatedAt;
    private String description;
    private List<String> labels = new ArrayList<>();
    private String projectId;
    private String sprintId;
}

File: frontend/src/types/issue.ts
import type { User } from './user';
export type IssueStatus = 'todo' | 'in-progress' | 'review' | 'done';
export type IssuePriority = 'highest' | 'high' | 'medium' | 'low' | 'lowest';
export interface Issue {
  id: string;
  key: string;
  title: string;
  status: IssueStatus;
  priority: IssuePriority;
  assignee: User | null;
  reporter: User;
  createdAt: string;
  updatedAt: string;
  description: string;
  labels: string[];
  projectId: string;
  sprintId?: string;
}

export interface Comment {
  id: string;
  issueId: string;
  author: User;
  content: string;
  createdAt: string;
}
